### PR TITLE
New auth module on SwiftUI with CTA

### PR DESCRIPTION
**Pull request by azamat163**

### CODE REVIEW

File: SmartReceipts/Services/APIs/AuthService.swift [hunk 1]
```diff
@@ -28,6 +28,9 @@ protocol AuthServiceInterface {
     func logout() -> Single<Void>
     func getUser() -> Single<User>
     func saveDevice(token: String) -> Single<Void>
+    
+    func asyncLogin(credentials: Credentials) async throws -> LoginResponse
```
Comment (EvsenevDev) [inline SmartReceipts/Services/APIs/AuthService.swift:32]: We can remove **async** from function name it pretty readable by signature of fucntion you can use just "login"

File: SmartReceipts/Services/APIs/Responses/AuthResponses.swift [hunk 2]
```diff
@@ -13,7 +13,21 @@ struct SignupResponse: Codable {
     private(set) var token: String
 }
 
+extension SignupResponse: Equatable {
+    static var empty = Self(
```
Comment (EvsenevDev) [inline SmartReceipts/Services/APIs/Responses/AuthResponses.swift:17]: Have you tried `empty = .init(id: "", token: "")` ?

File: SmartReceipts/SwiftUI/Modules/AuthModule/Client/AuthClient.swift [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    var login: (Credentials) async throws -> LoginResponse
+    var signup: (Credentials) async throws -> SignupResponse
+}
+
+extension AuthClient: DependencyKey {
+    static var liveValue = AuthClient(
+        login: { credentials in
+            try await AuthService.shared.asyncLogin(credentials: credentials)
+        },
+        signup: { credentials in
```
Comment (EvsenevDev) [inline SmartReceipts/SwiftUI/Modules/AuthModule/Client/AuthClient.swift:22]: I know that's just code style and up to you, but one line code like this looks more readable:

` signup: {  try await AuthService.shared.asyncSignup(credentials: $0) }`

File: SmartReceipts/SwiftUI/Modules/AuthModule/Views/AuthViewScreen.swift [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+                        .frame(maxWidth: .infinity, minHeight: 41)
+                        .background(Color(UIColor.srViolet))
+                        .clipShape(Capsule())
+                    }
+                    .fixedSize(horizontal: false, vertical: true)
+                    .padding(.top, 16)
+                    
+                    Spacer()
+                }
+                .toast(
```
Comment (EvsenevDev) [inline SmartReceipts/SwiftUI/Modules/AuthModule/Views/AuthViewScreen.swift:76]: Can we use just one toast and configre different state for it in Reducer?

File: SmartReceipts/Services/APIs/Responses/AuthResponses.swift (hunk 2 above)
Comment (azamat163) [reply SmartReceipts/Services/APIs/Responses/AuthResponses.swift:17]: убрал вовсе

**Review by EvsenevDev — APPROVED** [review]

**Pull request merged**